machine_learning_hep/optimization.py

```diff
@@ -34,15 +35,15 @@ def calc_efficiency(df_to_sel, sel_signal, name, num_step, sel_signal_std=None):
     eff_array = []
     eff_err_array = []
 
-    if sel_signal_std is None:
+    if calc_for_std is None:
         for thr in x_axis:
             num_sel_cand = len(df_to_sel[df_to_sel['y_test_prob' + name].values >= thr])
             eff = num_sel_cand / num_tot_cand
             eff_err = np.sqrt(eff * (1 - eff) / num_tot_cand)
             eff_array.append(eff)
             eff_err_array.append(eff_err)
     else:
-        num_sel_cand = len(df_to_sel.query(sel_signal_std))
+        num_sel_cand = len(df_to_sel)
```

Comment:
Ah you are right. The efficiency for the standard analysis will now always be one. I removed this query because of the same reason as I wrote above (for analyses with a bitmap selection, the sel_signal_std is a dummy variable that doesn't do anything anymore). This should be fixed, but not here, but in your third comment I think